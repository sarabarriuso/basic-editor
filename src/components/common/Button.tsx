import classnames from 'classnames';
import React from 'react';

export interface IButtonProps {
  onClick: (e: React.MouseEvent<HTMLButtonElement>) => void;
  text: string;
  className?: string;
  disabled?: boolean;
  variant?: ButtonVariant;
}
export enum ButtonVariant {
  Primary = 'Primary',
  Secondary = 'Secondary',
  Tertiary = 'Tertiary',
}

const Button: React.FC<IButtonProps> = ({
  className,
  onClick,
  text,
  disabled,
  variant = ButtonVariant.Primary,
}) => {
  const DEFAULT_CLASSNAME = 'button';

  const buttonClassName: string = classnames(DEFAULT_CLASSNAME, className, {
    [`${DEFAULT_CLASSNAME}--secondary`]: variant === ButtonVariant.Secondary,
    [`${DEFAULT_CLASSNAME}--tertiary`]: variant === ButtonVariant.Tertiary,
  });

  return (
    <button
      className={buttonClassName}
      type='submit'
      onClick={onClick}
      disabled={disabled}
    >
      {text}
    </button>
  );
};

export default Button;
