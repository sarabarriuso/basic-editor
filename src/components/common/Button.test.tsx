import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Button from './Button';

export default describe('Button Tests', () => {
  it('Renders Button successfully', () => {
    const tree = shallow(
      <Button
        className='testClass'
        onClick={(): void => {
          return;
        }}
        text='Example Text'
      />,
    );

    expect(toJson(tree)).toMatchSnapshot();
  });
});
