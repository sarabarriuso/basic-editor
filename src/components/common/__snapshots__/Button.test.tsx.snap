// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Tests Renders Button successfully 1`] = `
<button
  className="button testClass"
  onClick={[Function]}
  type="submit"
>
  Example Text
</button>
`;
